core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -74,6 +76,8 @@ class DeleteFileIndex {
   private final Map<Pair<Integer, StructLikeWrapper>, DeleteFileGroup> deletesByPartition;
   private final boolean isEmpty;
 
+  /** @deprecated since 1.4.0, will be removed in 1.5.0. */
+  @Deprecated
```

Comment:
Was only used in tests to construct an index from files, which is no longer needed.